unstructured/ingest/v2/processes/connectors/azure_cognitive_search.py

```diff
@@ -0,0 +1,225 @@
+            elements_contents = json.load(elements_file)
+
+        conformed_elements = [self.conform_dict(data=element) for element in elements_contents]
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(conformed_elements, output_file)
+        return output_path
+
+
+@dataclass
+class AzureCognitiveSearchUploader(Uploader):
+    upload_config: AzureCognitiveSearchUploaderConfig
+    connection_config: AzureCognitiveSearchConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    @DestinationConnectionError.wrap
+    @requires_dependencies(["azure"], extras="azure-cognitive-search")
+    def write_dict(self, *args, elements_dict: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        import azure.core.exceptions
+
+        logger.info(
+            f"writing {len(elements_dict)} documents to destination "
+            f"index at {self.connection_config.index}",
+        )
+        try:
+            results = self.connection_config.generate_client().upload_documents(
+                documents=elements_dict
+            )
+
+        except azure.core.exceptions.HttpResponseError as http_error:
+            raise WriteError(f"http error: {http_error}") from http_error
+        errors = []
+        success = []
+        for result in results:
+            if result.succeeded:
+                success.append(result)
+            else:
+                errors.append(result)
+        logger.debug(f"results: {len(success)} successes, {len(errors)} failures")
+        if errors:
+            raise WriteError(
+                ", ".join(
+                    [
+                        f"{error.key}: [{error.status_code}] {error.error_message}"
+                        for error in errors
+                    ],
+                ),
+            )
+
+    def write_dict_wrapper(self, elements_dict):
+        return self.write_dict(elements_dict=elements_dict)
+
+    def run(self, contents: list[UploadContent], **kwargs: t.Any) -> None:
+
+        elements_dict = []
+        for content in contents:
+            with open(content.path) as elements_file:
+                elements = json.load(elements_file)
+                elements_dict.extend(elements)
+
+        logger.info(
+            f"writing document batches to destination"
+            f" endpoint at {str(self.connection_config.endpoint)}"
+            f" index at {str(self.connection_config.index)}"
+            f" with batch size {str(self.upload_config.batch_size)} and"
+            f" {str(self.upload_config.num_processes)} (number of) processes"
+        )
+
+        batch_size = self.upload_config.batch_size
+
+        if self.upload_config.num_processes == 1:
+            for chunk in chunk_generator(elements_dict, batch_size):
+                self.write_dict(elements_dict=chunk)  # noqa: E203
+
+        else:
+            with mp.Pool(
```

Comment:
The original connector does not use multiprocessing. Are we sure we want to add it here?

Reply:
I now agree that other ways of parallelization might be better suited. Removed with: [remove multiprocessing](https://github.com/Unstructured-IO/unstructured/pull/3311/commits/c67307f9c57ec3d5b01cc0b8ec9906c8bd7f441a)